### PR TITLE
[Foundation] Add benchmarks for Data APIs

### PR DESCRIPTION
**Pull request by phausler**
This adds a number of benchmarks for Data

### CODE REVIEW

**Comment by phausler** [comment]
@swift-ci please benchmark

**Comment by phausler** [comment]
@swift-ci please smoke test

File: benchmark/single-source/DataBenchmarks.swift [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+    BenchmarkInfo(name: "Reset", runFunction: run_Reset, tags: [.validation, .api, .Data]),
+    BenchmarkInfo(name: "RepalceSmall", runFunction: run_RepalceSmall, tags: [.validation, .api, .Data]),
+    BenchmarkInfo(name: "ReplaceMedium", runFunction: run_ReplaceMedium, tags: [.validation, .api, .Data]),
+    BenchmarkInfo(name: "RepalceLarge", runFunction: run_RepalceLarge, tags: [.validation, .api, .Data]),
+    BenchmarkInfo(name: "RepalceSmallCountable", runFunction: run_RepalceSmallCountable, tags: [.validation, .api, .Data]),
+    BenchmarkInfo(name: "ReplaceMediumCountable", runFunction: run_ReplaceMediumCountable, tags: [.validation, .api, .Data]),
+    BenchmarkInfo(name: "RepalceLargeCountable", runFunction: run_RepalceLargeCountable, tags: [.validation, .api, .Data]),
+    BenchmarkInfo(name: "RepalceSmallBuffer", runFunction: run_RepalceSmallBuffer, tags: [.validation, .api, .Data]),
+    BenchmarkInfo(name: "ReplaceMediumBuffer", runFunction: run_ReplaceMediumBuffer, tags: [.validation, .api, .Data]),
+    BenchmarkInfo(name: "RepalceLargeBuffer", runFunction: run_RepalceLargeBuffer, tags: [.validation, .api, .Data]),
```
Comment (xwu) [inline benchmark/single-source/DataBenchmarks.swift]: A few typos here with the word "Replace" misspelt.

File: benchmark/utils/main.swift [hunk 2]
```diff
@@ -40,6 +40,7 @@ import CharacterProperties
 import Chars
 import ClassArrayGetter
 import Combos
+import DataBenchmarks
```
Comment (xwu) [inline benchmark/utils/main.swift]: Q: don't you also need to invoke `registerBenchmark` below?

**Comment by phausler** [comment]
@swift-ci please benchmark

**Comment by phausler** [comment]
@swift-ci please smoke test

**Review by airspeedswift — CHANGES_REQUESTED** [review]
I'm not sure what the situation is with sub-benchmark suites at the moment (it looks like all the string microbenchmarks that weren't intended to run by default on every PR are running). 

Need to be very careful to tune them to be stable when nothing changes, by doing enough iterations and also checking the min/max doesn't have too wide a variance.

**Comment by parkera** [comment]
Closing in favor of #15221.

**Pull request closed without merging**